### PR TITLE
elmPackages.*: add meta

### PR DESCRIPTION
**Pull request by turboMaCk**
###### Motivation for this change
`elmPackages` were missing meta attributes. This PR adds all missing meta information for rest of the tooling to consume. This will be handy especially as nixos-search will soon be able to filter packages by license, maintainer etc https://github.com/NixOS/nixos-search/pull/261.

In the future we should also add platform information.

###### Things done
Added meta.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @domenkozar (I've added you as a maintainer of elm compiler)

### CODE REVIEW

File: pkgs/development/compilers/elm/default.nix [hunk 1]
```diff
@@ -7,7 +7,7 @@ let
   fetchElmDeps = import ./fetchElmDeps.nix { inherit stdenv lib fetchurl; };
 
   hsPkgs = haskell.packages.ghc883.override {
-    overrides = self: super: with haskell.lib;
+    overrides = self: super: with haskell.lib; with stdenv.lib;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:10]:
```suggestion
    overrides = self: super: with haskell.lib; with lib;
```

by recent contribution change.

File: pkgs/development/compilers/elm/default.nix [hunk 2]
```diff
@@ -23,6 +23,11 @@ let
                 wrapProgram $out/bin/elm \
                   --prefix PATH ':' ${lib.makeBinPath [ nodejs ]}
               '';
+
+              description = "A delightful language for reliable webapps.";
+              homepage = "https://elm-lang.org/";
+              license = licenses.bsd3;
+              maintainers = [ maintainers.domenkozar maintainers.turbomack ];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:30]:
```suggestion
              maintainers = with maintainers; [ domenkozar turbomack ];
```

File: pkgs/development/compilers/elm/default.nix [hunk 3]
```diff
@@ -23,6 +23,11 @@ let
                 wrapProgram $out/bin/elm \
                   --prefix PATH ':' ${lib.makeBinPath [ nodejs ]}
               '';
+
+              description = "A delightful language for reliable webapps.";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:27]:
```suggestion
              description = "Delightful language for reliable webapps";
```

File: pkgs/development/compilers/elm/default.nix [hunk 4]
```diff
@@ -31,6 +36,11 @@ let
             */
             elm-format = justStaticExecutables (overrideCabal (self.callPackage ./packages/elm-format.nix {}) (drv: {
               jailbreak = true;
+
+              description = "elm-format formats Elm source code according to a standard set of rules based on the official Elm Style Guide";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:40]:
```suggestion
              description = "Formats Elm source code according to a standard set of rules based on the official Elm Style Guide";
```

File: pkgs/development/compilers/elm/default.nix [hunk 5]
```diff
@@ -31,6 +36,11 @@ let
...
+              homepage = "https://github.com/avh4/elm-format";
+              license = licenses.bsd3;
+              maintainers = [ maintainers.avh4 maintainers.turbomack ];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:43]:
```suggestion
              maintainers = with maintainers; [ avh4 turbomack ];
```

File: pkgs/development/compilers/elm/default.nix [hunk 6]
```diff
@@ -39,6 +49,11 @@ let
                 hpack
               '';
               jailbreak = true;
+
+              description = "This is a small tool that reads .elmi files (Elm interface file) generated by the elm compiler. These files contain information about exposed definitions (e.g. custom types, type aliases and functions).";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:53]: This description is a bit long and probably fits better into longDescription.

File: pkgs/development/compilers/elm/default.nix [hunk 7]
```diff
@@ -48,6 +63,11 @@ let
               jailbreak = true;
               # Tests are failing because of missing instances for Eq and Show type classes
               doCheck = false;
+
+              description = "Instrument Elm code as a preprocessing step. Meant to be using in conjunction with elm-coverage.";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:67]: description should not contain dots at the end which probably means two sentences are to long.

File: pkgs/development/compilers/elm/default.nix [hunk 8]
```diff
@@ -74,9 +94,17 @@ let
     inherit (hsPkgs.elmPkgs) elm;
   };
 
-  elmRustPackages = {
+  elmRustPackages =  {
     elm-json = import ./packages/elm-json.nix {
       inherit rustPlatform fetchurl openssl stdenv pkg-config;
+    } // {
+      meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:101]:
```suggestion
      meta = with lib; {
```

File: pkgs/development/compilers/elm/default.nix (hunk 5 above)
Comment (turboMaCk) [inline pkgs/development/compilers/elm/default.nix:43]: @avh4 just so you know - I'm adding you as a maintainer to elm-format. I believe in past you were declared directly in `elm-format.nix` file but I think it was overwritten by cabal2nix when the file was regenerated during some update. Defining this here means we can regenerate the file without loosing the information. Feel free to suggest change if you don't want to be declared as a maintainer anymore.

File: pkgs/development/compilers/elm/default.nix [hunk 9]
```diff
@@ -89,10 +117,24 @@ let
...
+            license = licenses.bsd3;
+            maintainers = [ maintainers.turbomack ];
+          };
+        };
 
       elm-verify-examples = patchBinwrap [elmi-to-json]
-        nodePkgs.elm-verify-examples;
+        nodePkgs.elm-verify-examples // {
+          meta = with stdenv.lib; {
+            description = "Verify examples in your docs.";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:132]:
```suggestion
            description = "Verify examples in your docs";
```

File: pkgs/development/compilers/elm/default.nix [hunk 10]
```diff
@@ -111,17 +153,51 @@ let
...
+            license = licenses.bsd3;
+            maintainers = [ maintainers.turbomack ];
+          };
         });
 
       create-elm-app = patchNpmElm (patchBinwrap [elmi-to-json]
-        nodePkgs.create-elm-app);
+        nodePkgs.create-elm-app) // {
+          meta = with stdenv.lib; {
+            description = "Create Elm apps with no build configuration.";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:167]:
```suggestion
            description = "Create Elm apps with no build configuration";
```

File: pkgs/development/compilers/elm/default.nix [hunk 11]
```diff
@@ -111,17 +153,51 @@ let
...
+            license = licenses.mit;
+            maintainers = [ maintainers.turbomack ];
+          };
+        };
 
       elm-review = patchBinwrap [elmRustPackages.elm-json]
-        nodePkgs.elm-review;
+        nodePkgs.elm-review // {
+          meta = with stdenv.lib; {
+            description = "Analyzes Elm projects, to help find mistakes before your users find them.";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/default.nix:177]:
```suggestion
            description = "Analyzes Elm projects, to help find mistakes before your users find them";
```

**Review by SuperSandro2000 — COMMENTED** [review]
I didn't comment every occurrence stdenv.lib and possible uses of with maintainers.

File: pkgs/development/compilers/elm/default.nix (hunk 6 above)
Comment (turboMaCk) [reply pkgs/development/compilers/elm/default.nix:53]: All the descriptions are coppied from official source (either github repository or package manager information). I was not sure if it's OK to use custom one or not.

File: pkgs/development/compilers/elm/default.nix (hunk 3 above)
Comment (turboMaCk) [reply pkgs/development/compilers/elm/default.nix:27]: This is taken from https://elm-lang.org/ - I'm not sure if it makes sense to change this.

File: pkgs/development/compilers/elm/default.nix (hunk 4 above)
Comment (turboMaCk) [reply pkgs/development/compilers/elm/default.nix:40]: I was thinking about this one myself.

File: pkgs/development/compilers/elm/default.nix (hunk 10 above)
Comment (turboMaCk) [reply pkgs/development/compilers/elm/default.nix:167]: thanks I missed this one. One thing I did to the original taglines taken from upstream was to remove `.` if it was single sentence :+1:

**Comment by turboMaCk** [comment]
@SuperSandro2000 I went through meta again and shorten the description. I think it would be still fair to wait for @domenkozar and @avh4 before we merge this since I'm declaring them as maintainers without consulting them first. Thanks for the review!

File: pkgs/development/compilers/elm/default.nix (hunk 3 above)
Comment (turboMaCk) [reply pkgs/development/compilers/elm/default.nix]: I've removed trailing `.` but kept `A` as it is the official tagline.

**Comment by turboMaCk** [comment]
It turned out Aaron is not in maintainers list - removing.

File: pkgs/development/compilers/elm/default.nix (hunk 5 above)
Comment (turboMaCk) [reply pkgs/development/compilers/elm/default.nix]: Update: turned out your profile is not in maintainer list so I removed you.

**Comment by avh4** [comment]
:+1: I'm fine either way.  (For the record, I still plan to submit a nixpkg PR each time there's a new elm-format version, as I've been doing for a while now.)

**Pull request merged**